Source/WebCore/css/calc/CSSCalcTree+Parser.cpp

```diff
@@ -702,6 +702,35 @@ static std::optional<TypedChild> consumeProgress(CSSParserTokenRange& tokens, in
     return TypedChild { makeChild(WTFMove(op), *outputType), *outputType };
 }
 
+static std::optional<TypedChild> consumeValueWithoutSimplifyingCalc(CSSParserTokenRange& tokens, int depth, ParserState& state)
+{
+    // Complex arguments need to be surrounded by a math function.
+    if (tokens.peek().type() == LeftParenthesisToken)
+        return { };
+
+    auto isFunction = !!tokens.peek().functionId();
+
+    auto typedValue = parseCalcValue(tokens, depth, state);
+    if (!typedValue)
+        return { };
+
+    auto isLeafValue = WTF::switchOn(typedValue->child,
+        [](Leaf auto&) { return true; },
+        [](auto&) { return false; }
+    );
+
+    if (isFunction && isLeafValue) {
+        // Wrap in Sum to keep top level calc() function in serialization.
+        Children children;
+        children.append(WTFMove(typedValue->child));
+        Sum sum { WTFMove(children) };
```

Comment:
Can just write `Sum { WTFMove(children) }` below in the expression; might read better?